docs/design/2018-10-08-online-DDL.md

```diff
@@ -0,0 +1,75 @@
+# TiDB DDL architecture
+- Author(s):     [zimuxia](https://github.com/zimulala) (Xia Li)
+- Last updated:  2018-10-08
+## Overview 
+This article will describe the architecture of the Online DDL implemented in TiDB.
+## Background
+Many databases lock tables when performing DDL operations. During this time, many transactions involved in this table are blocked (some databases support reading operations, but at the expense of consuming a large amount of memory). The larger the table is, the longer it affects the transactions.
+To simplify the design, the entire system allows only one node to make schema changes at the same time.
+* **TiDB server**: It contains TiDB's MySQL Protocol layer and TiDB SQL layer. Figure 1 mainly describes the basic modules involved in asynchronous schema changes at the TiDB SQL layer.
+* **load schema**: Creating a goroutine at the start of each node is in order to load the schema after each interval reached or a schema version update being found. The load failure here includes a load timeout.
+* **job**: Every DDL operation can be considered a job. At the beginning of a DDL operation, the operation is categorized into two types (general job and add index job) and encapsulated into one job to be stored into the general job queue or add index job queue. When this operation is completed, the job will be deleted from the corresponding job queue and stored in the history job queue, which is convenient for viewing historical jobs.
+* **general job queue**: It is a queue to store general jobs, staying at the KV layer. Logically, there is only one such queue for the entire system.
+* **add idx job queue**: It is a queue to store add index jobs, staying at the KV layer. Logically, there is only one such queue for the entire system.
+* **job history queue**: It is a queue to store the jobs that have been successfully processed, staying at the KV layer. Logically, there is only one such queue for the entire system.
```

Comment:
successfully processed? Just `have been processed` maybe better? Because the job meets errors or canceled will be enqueue to the history queue too.